src/DeploymentProvider/AzureSpringAppsDeploymentProvider.ts

```diff
@@ -21,14 +21,14 @@ export class AzureSpringAppsDeploymentProvider {
 
     public async preDeploymentStep() {
         const token = getDefaultAzureCredential();
-        this.client = new AppPlatformManagementClient(token, this.params.azureSubscription);
+        this.client = new asa.AppPlatformManagementClient(token, this.params.azureSubscription);
         const serviceList = await this.client.services.listBySubscription();
-        let filteredResources: Array<Models.ServiceResource> = [];
-        serviceList.forEach(service => {
+        let filteredResources: Array<asa.ServiceResource> = [];
+        for await (const service of serviceList) {
```

Comment:
Do we need `await` here?

Reply:
Yes. I think we need `await` here. The latest JS SDK has some changes to the return value of list result. For example, from `Array<ServiceResource>` to `PagedAsyncIterableIterator<ServiceResource>` Refer to this [link](
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html#the-for-await-of-statement), we need to use `for..await..of` statement to iterate over an async iterable

Reply:
I see. Thanks for the info.